server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -152,6 +156,9 @@ public class CompactionCoordinator implements CompactionCoordinatorService.Iface
   private static final Cache<ExternalCompactionId,RunningCompaction> COMPLETED =
       Caffeine.newBuilder().maximumSize(200).expireAfterWrite(10, TimeUnit.MINUTES).build();
 
+  private static final Cache<Path,Integer> CHECKED_TABLET_DIRS =
+      Caffeine.newBuilder().maximumSize(1000).build();
```

Comment:
The more of these we can cache the better for the NNs.  I think it would be good to use a weigher instead of a count and cache like 10MB of dir strings.

Reply:
ok. given that it's in the Manager, should we go larger? 50MB? 100MB?

Reply:
So, with the Weigher you don't get eviction based on recency. It just starts evicting entries when it's over the size limit.

Reply:
> So, with the Weigher you don't get eviction based on recency. It just starts evicting entries when it's over the size limit.

What documentation where you looking at? I found [these docs](https://github.com/ben-manes/caffeine/wiki/Eviction#size-based) which seem to indicate when cache is over size it evicts based on how recently used it was.

Reply:
Ok, so I may have misread the docs.  :-)

Reply:
Modified to use Weigher in 585f843